src/index.ts

```diff
@@ -275,7 +279,12 @@ export function register (options: Options = {}): Register {
             return undefined
           }
 
-          cache.contents[fileName] = getFile(fileName)
+          const content = getFile(fileName)
+          if (shouldVerifyCache && !isValidCacheContent(content)) {
```

Comment:
This isn't coming from the cache, it's coming from your original source file (e.g. this is your TypeScript, so should never pass this check).